modules/builds-running-entitled-builds-with-sharedsecret-objects.adoc

```diff
@@ -0,0 +1,144 @@
+:_mod-docs-content-type: PROCEDURE
+[id="builds-running-entitled-builds-with-sharedsecret-objects_{context}"]
+= Running builds using SharedSecret objects
+
+You can use a `SharedSecret` object to securely access the entitlement keys of a cluster in builds.
+
+The `SharedSecret` object allows you to share and synchronize secrets across namespaces.
+
+[IMPORTANT]
+====
+The Shared Resource CSI Driver feature is now generally available in link:https://docs.redhat.com/en/documentation/builds_for_red_hat_openshift/1.1[{builds-v2title} 1.1]. This feature is now deprecated in {product-title}. To use this feature, ensure you are using {builds-v2title} 1.1 or a more recent version.
+====
+
+.Prerequisites
+
+* You must have permission to perform the following actions:
+** Create build configs and start builds.
+** Discover which `SharedSecret` custom resource (CR) instances are available by entering the `oc get sharedsecrets` command and getting a non-empty list back.
+** Determine if the `builder` service account available to you in your namespace is allowed to use the given `SharedSecret` CR instance. In other words, you can run `oc adm policy who-can use <identifier_of_specific_SharedSecret>` to see if the `builder` service account in your namespace is listed.
+
+[NOTE]
+====
+If neither of the last two prerequisites in this list are met, establish, or ask someone to establish, the necessary role-based access control (RBAC) so that you can discover `SharedSecret` CR instances and enable service accounts to use `SharedSecret` CR instances.
+====
+
+.Procedure
+
+. Create a `SharedSecret` object instance with the cluster's entitlement secret by running the `oc apply` command:
++
+[IMPORTANT]
+====
+You must have cluster administrator permissions to create `SharedSecret` objects.
+====
++
+[source,terminal]
+----
+$ oc apply -f -<<EOF
+apiVersion: sharedresource.openshift.io/v1alpha1
+kind: SharedSecret
+metadata:
+  name: share-rhel-entitlement
+spec:
+  secretRef:
+    name: etc-pki-entitlement
+    namespace: openshift-config-managed
+EOF
+----
+
+. Create a role to grant permission to access the `SharedSecret` object and a `RoleBinding` object that grants permission to access the `SharedSecret` object by running the following command:
++
+[source,terminal]
+----
+$ oc new-project buildah-entitlement
+$ oc apply -f -<<EOF
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: use-shared-secret
+rules:
+  - apiGroups:
+      - sharedresource.openshift.io
+    resources:
+      - sharedsecrets
+    resourceNames:
+      - share-entitlement-secret
+    verbs:
+      - use
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: use-shared-secret
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: use-shared-secret
+subjects:
+  - kind: ServiceAccount
+    name: pipeline
+    namespace: buildah-entitlement
+EOF
+----
```

Comment:
We are missing a step from [BUILD-965](https://issues.redhat.com/browse/BUILD-965) - to get us to GA, we reduced the default set of permissions granted to the CSI driver's service account. Admins need to explicitly grant the driver permission to access a given shared resource.

The procedure described in https://github.com/openshift/csi-driver-shared-resource is a bit too permissive, particularly for this use case. We can do better, and I'll try and post an example.